src/tool/subcommands/api_cmd.rs

```diff
@@ -671,6 +717,237 @@ async fn compare_apis(
+            .with_method(STATE_MINER_ACTIVE_SECTORS, state_miner_active_sectors::<DB>)
+            .with_method(STATE_MINER_SECTOR_COUNT, state_miner_sector_count::<DB>)
+            .with_method(STATE_MINER_FAULTS, state_miner_faults::<DB>)
+            .with_method(STATE_MINER_RECOVERIES, state_miner_recoveries::<DB>)
+            .with_method(STATE_MINER_POWER, state_miner_power::<DB>)
+            .with_method(STATE_MINER_DEADLINES, state_miner_deadlines::<DB>)
+            .with_method(STATE_LIST_MINERS, state_list_miners::<DB>)
+            .with_method(
+                STATE_MINER_PROVING_DEADLINE,
+                state_miner_proving_deadline::<DB>,
+            )
+            .with_method(STATE_GET_RECEIPT, state_get_receipt::<DB>)
+            .with_method(STATE_WAIT_MSG, state_wait_msg::<DB>)
+            .with_method(STATE_SEARCH_MSG, state_search_msg::<DB>)
+            .with_method(STATE_SEARCH_MSG_LIMITED, state_search_msg_limited::<DB>)
+            .with_method(STATE_FETCH_ROOT, state_fetch_root::<DB>)
+            .with_method(
+                STATE_GET_RANDOMNESS_FROM_TICKETS,
+                state_get_randomness_from_tickets::<DB>,
+            )
+            .with_method(
+                STATE_GET_RANDOMNESS_FROM_BEACON,
+                state_get_randomness_from_beacon::<DB>,
+            )
+            .with_method(STATE_READ_STATE, state_read_state::<DB>)
+            .with_method(STATE_CIRCULATING_SUPPLY, state_circulating_supply::<DB>)
+            .with_method(STATE_SECTOR_GET_INFO, state_sector_get_info::<DB>)
+            .with_method(
+                STATE_VM_CIRCULATING_SUPPLY_INTERNAL,
+                state_vm_circulating_supply_internal::<DB>,
+            )
+            .with_method(MSIG_GET_AVAILABLE_BALANCE, msig_get_available_balance::<DB>)
+            .with_method(MSIG_GET_PENDING, msig_get_pending::<DB>)
+            // Gas API
+            .with_method(GAS_ESTIMATE_FEE_CAP, gas_estimate_fee_cap::<DB>)
+            .with_method(GAS_ESTIMATE_GAS_LIMIT, gas_estimate_gas_limit::<DB>)
+            .with_method(GAS_ESTIMATE_GAS_PREMIUM, gas_estimate_gas_premium::<DB>)
+            .with_method(GAS_ESTIMATE_MESSAGE_GAS, gas_estimate_message_gas::<DB>)
+            // Common API
+            .with_method(VERSION, move || {
+                version(block_delay, FOREST_VERSION_STRING.as_str())
+            })
+            .with_method(SESSION, session)
+            .with_method(START_TIME, start_time::<DB>)
+            // Node API
+            .with_method(NODE_STATUS, node_status::<DB>)
+            // Eth API
+            .with_method(ETH_ACCOUNTS, eth_accounts)
+            .with_method(ETH_BLOCK_NUMBER, eth_block_number::<DB>)
+            .with_method(ETH_CHAIN_ID, eth_chain_id::<DB>)
+            .with_method(ETH_GAS_PRICE, eth_gas_price::<DB>)
+            .with_method(ETH_GET_BALANCE, eth_get_balance::<DB>)
+            .finish_unwrapped(),
```

Comment:
have I seen this code before? Could we refactor?

Reply:
Yeah, it would be really great to have a single "helper" method for all the occurrences of RPC definitions, assuming they are all the same. If not - perhaps whatever is shared is a good start.

Something like:

`fn rpc_methods(server: Arc<Server>)` or otherwise we could define a custom trait and implement it for the `Server` struct, see [extension traits](https://rust-lang.github.io/rfcs/0445-extension-trait-conventions.html) for inspiration.

Reply:
Ah, I can see now that this is RPC `v1`. As opposed to `v0` or `ws`. Curious if there is already a definition for that elsewhere.